test/server.cpp

```diff
@@ -1013,15 +961,42 @@ TEST_F(ServerTest, RandomPageRedirectsToAnExistingArticle)
   ASSERT_TRUE(kiwix::startsWith(g->get_header_value("Location"), "/ROOT/content/zimfile/A/"));
 }
 
-TEST_F(ServerTest, BookMainPageIsRedirectedToArticleIndex)
+TEST_F(ServerTest, NonEdpointUrlsAreRedirectedToContentUrls)
 {
   {
   auto g = zfs1_->GET("/ROOT/zimfile");
   ASSERT_EQ(302, g->status);
   ASSERT_TRUE(g->has_header("Location"));
+  ASSERT_EQ("/ROOT/content/zimfile", g->get_header_value("Location"));
+  }
+
+  {
+  auto g = zfs1_->GET("/ROOT/zimfile/A/index");
+  ASSERT_EQ(302, g->status);
+  ASSERT_TRUE(g->has_header("Location"));
   ASSERT_EQ("/ROOT/content/zimfile/A/index", g->get_header_value("Location"));
   }
 
+  {
+  // XXX: Should this be a redirect or we better return an error immediately
+  // XXX: when the target URL is invalid?
```

Comment:
> The error is because sever may not contain the requested book.

I wrote that comment because of a concern about a different case - when the top-level component is **not** intended to designate a book. Then the text of the 404 Not Found page can be somewhat misleading. For example:

- usage of an obsolete endpoint (e.g. `/meta`, converted to `/content/meta` in the 404 error)
- a typo in the manually typed API endpoint (e.g. `/ramdon`, converted to `/content/ramdon` in the 404 error)
- usage of a non-empty root location for a kiwix-server started without the `--urlRootLocation` option (`/ROOT/random` is converted to `/content/ROOT/random` in the 404 error)

Anyway, I've removed that comment for now.